bots/mainbot/castle.js

```diff
@@ -1,172 +1,352 @@
 'use strict';
 
-import {BCAbstractRobot, SPECS} from 'battlecode';
+import { BCAbstractRobot, SPECS} from 'battlecode';
 import nav from './nav.js' ;
 import utilities from './utilities.js'
 import combat from './combat.js'
 
 const castle = {};
 const pilgrims ={};
-
-
 pilgrims.number = 0;
-
-var first_castle = true ;
-var castle_locs = [];
-
-
+var castle_ids = [];
+var builds = 0;
+var buildunitflag =0 ;
 castle.takeTurn = (self) => {
-	self.log('castle taking turn')
 	
+	self.log('castle taking turn')
 	var robotsnearme = self.getVisibleRobots();
 
-	var getBuildDir = function(buildunit) {
-        var options = nav.rotate_arr.filter((d) => {
-            return nav.isPassable(nav.applyDir(self.me, d), self.getPassableMap(), self.getVisibleRobotMap())
-        })
-        return options[0];
-	}
-
+			
 	var attackable = robotsnearme.filter((r) => {
```

Comment:
For next spring we might want to clean up some of this code. There is already a function in the utilities file to get a list of robots in your attack range.

Reply:
I did try to use the existing functionality from utilities but for some reason it was not attacking accurately hence used this instead.
Once all your changes are merged i can use the same and try again